### PR TITLE
Port multiple signature support from saml2tu

### PR DESCRIPTION
**Pull request by mgduk**
As the saml2tu repo isn't a true fork (it has a single commit of all the saml2
history) there's no simple way to merge it in. I've manually ported the
relevant code change it introduced, including the test for it.

The behavioral change this introduces to the saml2 repo:

Previous behavior:
If there are multiple signatures in a SAML document, it will be rejected.

New behavior:
If there are multiple signatures, the first will be checked and used if valid.
Other signatures beyond the first will be ignored.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,7 +31,7 @@
     "async": "~1.5.2",
     "debug": "^1.0.4",
     "underscore": "~1.6.0",
-    "xml-crypto": "^0.8.1",
+    "xml-crypto": "^0.8.2",
```
Comment (brettkiefer) [inline package.json]: Are we sure we need this?

Comment (mgduk) [inline package.json]: Reverted to v0.8.1 as saml2-js has

**Comment by brettkiefer** [comment]
Squash and 👍

**Comment by mgduk** [comment]
For reference, saml2tu switched to xml-crypto 0.8.2 as part of [“performing some final integration tests against our SAML callback server”](https://github.com/bcoe/saml2tu/commit/f874c9c104f77b9c7629e7008f566bd737197f06) so I'm guessing it fixed some other bug they were seeing, but I can't see how it's directly relevant to the change to use the first of multiple sigs.

**Pull request merged**